pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -501,16 +518,16 @@ private T decode(byte[] schemaVersion) {
         }
     }
 
-    private T decodeBySchema(byte[] schemaVersion) {
-        T value = poolMessage ? schema.decode(payload.nioBuffer(), schemaVersion) : null;
+    private T decodeBySchema(byte[] schemaFlag) {
```

Comment:
Should it be renamed with `schemaId`? The `schemaFlag` concept in code is confusing. I see the result of `getSchemaFlag()` is still passed to `getSchemaInfo` method, which accepts a schema version

Reply:
Changed get schemaInfo to use the schema version, changed the parameter name to `schemaIdOrVersion`.